src/uitext.ml

```diff
@@ -1163,19 +1163,40 @@ let displayWaitMessage () =
   if not (Prefs.read silent) then
     Util.msg "%s\n" (Uicommon.contactingServerMsg ())
 
+(* Most modern VT100 terminal emulators (and some ANSI) are able to switch
+   automatic line-wrapping off and on by control sequences ESC[?7l and ESC[?7h.
+   This here is a very blunt heuristic to filter out some that can't do it or
+   use a different control sequence. It does not need to be exact, as long as
+   it covers the vast majority of supported systems. *)
+let termNowrapOk =
+  System.termVtCapable Unix.stdout &&
+  let s = try System.getenv "TERM" with Not_found -> "" in
+  not (
+    s = "dumb"
+    || s = "emacs"
+    || Util.startswith s "sun"
+    || Util.startswith s "cons"
+    || Util.startswith s "eterm"
+    || Util.startswith s "cygwin"
+    || Util.startswith s "dvtm"
+  )
+
 let synchronizeOnce ?wantWatcher pathsOpt =
   let showStatus path =
     if path = "" then Util.set_infos "" else
-    let max_len = 70 in
-    let mid = (max_len - 3) / 2 in
-    let path =
+    let shorten path =
+      let max_len = 70 in
+      let mid = (max_len - 3) / 2 in
       let l = String.length path in
       if l <= max_len then path else
       String.sub path 0 (max_len - mid - 3) ^ "..." ^
       String.sub path (l - mid) mid
     in
     let c = "-\\|/".[truncate (mod_float (4. *. Unix.gettimeofday ()) 4.)] in
-    Util.set_infos (Format.sprintf "%c %s" c path)
+    if termNowrapOk && not (Prefs.read dumbtty) then
+      Util.set_infos (Format.sprintf "%c \027[?7l%s\027[?7h" c path) ~clr:"\r\027[K\r"
```

Comment:
Shouldn't the escape codes be coming from a capability?   It's not just "terminal emulators" but terminals, really, and there are a lot of termcap/terminfo things out there.

Reply:
> Shouldn't the escape codes be coming from a capability?

Ideally, yes. In practice, here, I don't think it matters. It appears that pretty much anything modern emulates VT100/xterm and supports these sequences (or at the very least silently ignores them). Some exceptions I've explicitly brought out above.

I thought about it but I don't want to make terminfo, termcap or some other thing a dependency (because it would have to be something like ncurses or s-lang then). It's just not worth it.

I did query the terminfo database and I'm satisfied that the current approach is not going to cause any harm.

> It's not just "terminal emulators" but terminals, really, and there are a lot of termcap/terminfo things out there.

I don't think we need to specially support actual terminals (which ones do you have in mind?). Also, any funny stuff is turned off by `TERM=dumb` or `-dumbtty`.